Content.Server/GameObjects/Components/Construction/ConstructionComponent.cs

```diff
@@ -141,46 +142,42 @@ bool TryProcessStep(ConstructionStep step, IEntity slapped)
                         sound.Play("/Audio/items/deconstruct.ogg", Transform.GridPosition);
                     return true;
                 case ConstructionStepTool toolStep:
+                    if (!slapped.TryGetComponent<ToolComponent>(out var tool))
+                        return false;
                     switch (toolStep.Tool)
```

Comment:
Here, you do a lot of `if (tool.Behavior ==`. You could factor it out and do `if (tool.Behavior == toolStep.Tool)` before the `switch`.

This way, you could factor some cases, like:
```c#
case Tool.CrowBar:
case Tool.Wrench:
case Tool.Wirecutters:
    tool.PlayUseSound();
    return true;
```

Alternatively, if you add TryCrowBar, TryWrench, TryWireCutters, ... and check if the current Behavior is the right tool in this methods, (like you already to in TryWeld, for instance), you can just do:
```c#
switch (toolStep.Tool)
{
case ToolType.CrowBar:
    if (tool.TryCrowBar())
    {
        tool.PlayUseSound();
        return true;
    }
    return false;
// Repeat for all tools, whithout need to check for tool.Behavior
}

Reply:
You're right, I could have cleaned up that code. I only made it compatible with the tool refactor without taking into account cleanup opportunities like these.
I cleaned it up now.

> Alternatively, if you add TryCrowBar, TryWrench, TryWireCutters, ... and check if the current Behavior is the right tool in this methods, (like you already to in TryWeld, for instance), you can just do [...]

I'm against that, TryWeld is only used to check if the welder has enough fuel to weld something. For the rest of these, you can just compare the Behavior like on your first suggestion, which I ended up doing.